be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala

```diff
@@ -41,9 +41,50 @@ class RollCallHandler(dbRef: => AskableActorRef) extends MessageHandler {
 
   def handleCreateRollCall(rpcRequest: JsonRpcRequest): GraphMessage = {
     val ask: Future[GraphMessage] = dbAskWritePropagate(rpcRequest)
-    Await.result(ask, duration)
+    Await.result(ask, duration) match {
+      case Left(_) =>
+        rpcRequest.getParamsMessage match {
+          case Some(message: Message) =>
+            val data: CreateRollCall = message.decodedData.get.asInstanceOf[CreateRollCall]
+            // we are using the rollcall id instead of the message_id at rollcall creation
+            val rollCallChannel: Channel = Channel(s"${Channel.ROOT_CHANNEL_PREFIX}${data.id}")
+            val combined = for {
+              // check whether the rollcall already exists in db
+              _ <- {
+                (dbActor ? DbActor.ChannelExists(rollCallChannel)).transformWith {
+                  case Success(_) => Future {
+                    throw DbActorNAckException(ErrorCodes.INVALID_ACTION.id, "rollcall already exists in db")
+                  }
+                  case _ => Future {
+                    ()
+                  }
+                }
+              }
+              // create rollcall channel
+              _ <- dbActor ? DbActor.CreateChannel(rollCallChannel, ObjectType.ROLL_CALL)
+              // write rollcall creation message
+              _ <- dbActor ? DbActor.WriteAndPropagate(rollCallChannel, message)
```

Comment:
This channel (roll call channel) is internal to our back-end. Basically, this will be undefined behaviour since you are sending the creation message to front-end twice but on different channels (lines 43 and 66)